### PR TITLE
fix: ensures `grantDonationListener` is called and successfully saves contribution data

### PR DESCRIPTION
**Pull request by gdixon**
This PR ensures that the `grantDonationListener` is called and that the data is correctly stored in to localForage/indexedDB. 

--

Fixes: #552

### CODE REVIEW

File: app/src/utils/data/contributions.ts [hunk 1]
```diff
@@ -308,7 +309,7 @@ export function grantDonationListener(
           inRounds: grantRounds,
           grantAddress: args.grantPayees[grantId],
           address: tx.from,
-          donationToken: args.donationToken,
+          donationToken: { ...args.donationToken },
```
Comment (gdixon) [inline app/src/utils/data/contributions.ts:312]: Attempting to save vue wrapped objects/primitives into `indexedDB` throws some nasty errors!

File: app/src/utils/data/contributions.ts [hunk 2]
```diff
@@ -361,7 +362,7 @@ export function grantDonationListener(
   // filter for every event by topic so that we can get to the event args too
   const filter = {
     address: GRANT_ROUND_MANAGER_ADDRESS,
-    topics: [id('GrantDonation(uint96,address,uint256,address[])')],
+    topics: [id('GrantDonation(uint96,address,uint256,address[],uint256)')],
```
Comment (gdixon) [inline app/src/utils/data/contributions.ts:365]: This adds the `time` arg to the topic filter 👍 - This is the main thing which was preventing the `GrantDonations` from updating.

**Review by thelostone-mc — APPROVED** [review]
Ooo tested locally ! works chil

Also lol on having the filter listener misconfigured XD

**Review by mk1r — APPROVED** [review]

**Comment by mk1r** [comment]
good fix, working well!

**Pull request merged**